### DISCUSSION TITLE
Statiq.Docs Roadmap

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [statiqdev/Discussions#114] [General]
I'm running into a use case where I'd like a code documentation site and staying with Statiq for this is an attractive prospect, but I understand that the Docs library is in need of a rework.

Is there a proper roadmap available at this point or would you invite work being done to establish one? Having read some of your comments there, @daveaglick, it sounds like this was either ported from Wyam and is still relying on some of its code or it's using a much older version of the Framework library (I've not actually dug into the code to look).

If you could sum up exactly where we are on that, I can look at working with you to create a roadmap for release and then start laying down some code to bring it up to date or rewrite it from scratch, whichever you feel would be most appropriate. I do feel we should create a branch in the Statiq.Docs library just for this though since it sounds like it'll be a more involved change. At that point we can begin closing or prioritizing issues as appropriate.

**Comment by daveaglick** [comment]
It's under very active development right now. Most of the engine and code bits are in place, so the bulk of the work is porting over the API documentation theme that makes use of the Statiq Docs pipelines and code analysis.

Here's where we're at today:
- [Statiq Docs](https://github.com/statiqdev/Statiq.Docs) is coming along nicely. That contains pipelines that perform code analysis on source files or assemblies and makes documents representing code analysis symbols available. It's mostly feature-complete at this point and is integrated into my normal release process now so it goes out as well when Statiq Framework or Statiq Web is updated since it depends on them (11 releases of Statiq Docs and counting). Unfortunately it doesn't do much for anyone without a corresponding theme - you could also make use of the `Api` pipeline outputs yourself in a bespoke template, but there's a lot to that so it's a heavy lift without an out-of-the-box theme to start with.
- [Docable](https://github.com/statiqdev/Docable) is the under-development theme for use with Statiq Docs. It says there hasn't been a commit since late September, but I've been working on it regularly, just haven't gotten to the point of checking anything else in yet. Should be a big drop over the holidays though. Most of the work here revolves around bringing over code from the Wyam API docs theme and fixing/refactoring it. There's also a fair amount of cascading changes that I've encountered - stuff that needs to change in either Statiq Framework or Statiq Web to accommodate features in the theme. That's expected and is the entire reason neither of those projects have gone 1.0 yet - I want to finish up the Statiq Docs work in case there are additional breaking changes needed upstream and then 1.0 everything at once.

Unfortunately this work is interrelated enough that it's mostly just me grinding. Getting all the bits to work smoothly in a unified development environment where I can work on all the projects at once [was tricky](https://github.com/statiqdev/Statiq) but now that I've up and running it's going pretty quickly. Bascially the only thingg left on any sort of roadmap right now is "get the theme finished" :).

**Reply by TheBrambleShark** [reply]
Sounds like it's a lot further along than I'd anticipated. Good to know!

I'm sure I'll be digging into things once that update is pushed to see what I'll want to do for my environment. That'll likely come in the form of theme modifications and the like, but having a starting place will definitely be appreciated.

In the mean time, I can go ahead and start tackling some of the smaller items after you push your update here.
